firstvoices/backend/resources/utils/import_export_widgets.py

```diff
@@ -113,51 +111,56 @@ def clean(self, value, row=None, **kwargs):
 
     def clean(self, value, row=None, **kwargs):
-        category_column_name_pattern = r"category[_2-5]*"
+        value = value.strip().lower()
 
-        input_categories = {}
-        valid_categories = []
+        if self.title_case:
+            value = value.title()
 
-        for column, input_value in row.items():
-            if re.fullmatch(category_column_name_pattern, column):
-                input_categories[column] = input_value.strip()
+        return super().clean(value, row, **kwargs)
 
-        # removing empty values from dict
-        input_categories = {
-            category: value for category, value in input_categories.items() if value
-        }
 
-        # If no categories provided, return
-        if len(input_categories) == 0:
-            return Category.objects.none()
+class CustomManyToManyWidget(ManyToManyWidget):
+    def __init__(self, model, column_name, field=None, *args, **kwargs):
+        self.model = model
+        self.field = field
+        self.column_name = column_name
+        super().__init__(model=self.model, field=field, *args, **kwargs)
 
-        # Validate categories
-        for column, input_value in input_categories.items():
-            category_lookup = Category.objects.filter(
-                site__id=row["site"], title=input_value
-            )
-            if len(category_lookup) == 0:
-                raise ValidationError(f"Invalid category supplied in column {column}.")
-            else:
-                valid_categories.append(category_lookup[0])
+    def clean(self, value, row=None, **kwargs):
+        column_name_pattern = rf"{self.column_name}[_2-5]*"
 
-        return valid_categories
+        input_entries = {}
+        valid_entries = []
 
+        for column, input_value in row.items():
+            if re.fullmatch(column_name_pattern, column):
+                input_entries[column] = input_value.strip()
 
-class CleanForeignKeyWidget(ForeignKeyWidget):
-    def __init__(self, model, field, title_case=False, *args, **kwargs):
-        self.title_case = title_case
-        super().__init__(model=model, field=field, *args, **kwargs)
+        # removing empty values from dict
+        input_entries = {
+            entry: value for entry, value in input_entries.items() if value
+        }
 
-    def clean(self, value, row=None, **kwargs):
-        value = value.strip().lower()
+        # If no entries provided, return
+        if len(input_entries) == 0:
+            return self.model.objects.none()
 
-        if self.title_case:
-            value = value.title()
+        # Validate entries
+        for column, input_value in input_entries.items():
+            entry_lookup = self.model.objects.filter(
+                site__id=row["site"], **{self.field: input_value}
+            )
+            if len(entry_lookup) == 0:
+                raise ValidationError(
+                    f"Invalid {str(self.model).lower()} supplied in column {column}."
```

Comment:
It would probably be helpful to list the expected field in this message as well.